### ISSUE TITLE
Boolean Offline Status

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#9020]
This issue is intended to be the sole issue associated with the [Boolean Offline Status design doc](https://docs.google.com/document/d/1r9tM7zRRC12gUWPibo_O0sZgyYxoogQNvrKz14SyEds/edit#).

Please reference the document for the complete problem statement and detailed implementation plan.

### Problem
Parts of our app are reliant upon a 3rd-party service called Pusher to communicate with our servers in real-time. There are a couple scenarios where the app may have an internet connection but still be unable to receive data from Pusher, for example:

- You are connected to the Pusher service but not subscribed to your user channel
- The Pusher service is down

In these scenarios, we currently have an inconsistent UX, where those parts of our app that are reliant upon Pusher (eg. "user is typing", comments being edited, etc.) won't work as expected, while others will. This will be confusing for users.

### Solution
Implement Boolean Offline Status according to the following logic:

**If you are authenticated with Expensify (have an authToken), you are considered online if and only if:**
- You have an internet connection (determined by NetInfo), AND
- You are connected to Pusher, AND
- You are subscribed to your private user Pusher channel

**If you are NOT authenticated with Expensify (do not have an authToken), you are considered online if and only if:**
- You have an internet connection (determined by NetInfo), AND
- You are connected to Pusher

**If you are not online, you’re offline. Period.**

**Comment by JmillsExpensify** [comment]
Commenting so this hits my participating filter.

**Comment by roryabraham** [comment]
Pro-tip – you can also just press `Subscribe`/`Unsubscribe` on the side 😄 

<img width="338" alt="image" src="https://user-images.githubusercontent.com/47436092/168870680-71091e30-368a-485c-be60-8cdc9d61bf92.png">

**Comment by roryabraham** [comment]
Putting this on HOLD per [this slack thread](https://expensify.slack.com/archives/C02HWMSMZEC/p1652799118955609)

**Comment by JmillsExpensify** [comment]
@roryabraham Yeah unfortunately that doesn't work for the "participating" filter in my GH inbox. Thanks though!

**Comment by JmillsExpensify** [comment]
More importantly though, are we still on hold?

**Comment by roryabraham** [comment]
Got a draft PR up, just need to spend a bit more time testing it and working on the PR writeup before it will be ready for review: https://github.com/Expensify/App/pull/9230/files

**Comment by roryabraham** [comment]
Going to prioritize this over the next few days. Updated the PR on Friday and was fighting Storybook, of all things.

**Comment by roryabraham** [comment]
Prioritized Comp tool improvements last week due to the deadline being today. Picking this back up this week

**Comment by roryabraham** [comment]
No update. This really only covers a sort of rare edge case where pusher or our web servers are down, so despite being under the Offline-First umbrella it's really not as high priority as some other items.

**Comment by roryabraham** [comment]
Speak of the devil: https://expensify.slack.com/archives/C03SC9DVD/p1659625927469579

**Comment by roryabraham** [comment]
Not prioritizing over other N7-blocking initiatives

**Comment by roryabraham** [comment]
Still not prioritizing over other N7 issues

**Comment by roryabraham** [comment]
Still not prioritizing, though a tangentially-related issue has cropped up [here](https://github.com/Expensify/App/issues/10524)

**Comment by arielgreen** [comment]
@roryabraham I'm going to swap this to `NewFeature`. Let me know if you disagree.

**Comment by roryabraham** [comment]
We could, but I like keeping it open as a monthly for now. I also think that implementing this will be a bit simpler once [this proposal](https://expensify.slack.com/archives/CC7NECV4L/p1666999594932679) is implemented

**Comment by JmillsExpensify** [comment]
Ok, cool. Thinking more on this, you're right. We still need to do this. I was also probably too wrapped up in the specific implementation of boolean offline status, and where it fits. Assigning myself as well to keep it top of mind.

**Comment by JmillsExpensify** [comment]
Added the WAQ hold for clarity.

**Comment by JmillsExpensify** [comment]
Still on hold for WAQ.

**Comment by JmillsExpensify** [comment]
No longer on hold for WAQ, but still not a priority in my book.

**Comment by roryabraham** [comment]
No update

**Comment by JmillsExpensify** [comment]
Out of curiosity, I wonder if we should consider this one for the "buddy" process?

**Comment by roryabraham** [comment]
Chatted with @JmillsExpensify, we're just going to close this for now
